src/diff/index.js

```diff
@@ -368,9 +368,11 @@ function diffElementNodes(
 			if (
 				'value' in newProps &&
 				newProps.value !== undefined &&
-				newProps.value !== dom.value
+				newProps.value !== dom.value &&
+				newProps.defaultValue !== newProps.value
 			) {
 				dom.value = newProps.value == null ? '' : newProps.value;
+				delete newProps.defaultValue; // is this needed?
```

Comment:
This shouldn't be needed since there are new props instances every time either way